src/gfp.h

```diff
@@ -0,0 +1,22 @@
+#ifndef BN256_GFP_H
+#define BN256_GFP_H
+
+#include <cstdint>
+#include <array>
+
+namespace bn256 {
+
+    typedef std::array<uint64_t, 4> uint64_array_4_t;
+    typedef uint64_array_4_t gfp;
```

Comment:
`gfp` allows for aggregate initialization which is nice, but we need to be careful with default initialization as those will be of indeterminate value. I didn't any issue with this, just a reminder to keep it in mind.